fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/identifiers/FedoraID.java

```diff
@@ -156,6 +188,88 @@ public String getMementoString() {
         return mementoDatetimeStr;
     }
 
+    /**
+     * Create a new FedoraID using the resource ID as the base
+     * @param addition One or more additional strings.
+     * @return new FedoraID for the new identifier.
+     */
+    public FedoraID addToResourceId(final String... addition) {
+        return FedoraID.create(this.getResourceId(), addition);
+    }
+
+    /**
+     * Create a new FedoraID using the full ID as the base.
+     * @param addition One or more additional strings.
+     * @return new FedoraID for the new identifier.
+     */
+    public FedoraID addToFullId(final String... addition) {
+        return FedoraID.create(this.getFullId(), addition);
+    }
+
+    @Override
+    public boolean equals(final Object obj) {
+        if (obj == this) {
+            return true;
+        }
+
+        if (!(obj instanceof FedoraID)) {
+            return false;
+        }
+
+        final var testObj = (FedoraID) obj;
+        return Objects.equals(testObj.getFullId(), this.getFullId());
+    }
+
+    @Override
+    public int hashCode() {
+        return getFullId().hashCode();
+    }
+
+    /**
+     * Takes one or more strings and combines them with /s
+     * @param firstPart The first part of the id, if blank return a zero length string.
+     * @param parts Zero or more additional string.
+     * @return A string of all parts with combined with /s
+     */
+    private static String idBuilder(final String firstPart, final String... parts) {
+        if (firstPart == null) {
+            return "";
+        }
+        final String[] allParts = Stream.of(new String[]{firstPart}, parts).flatMap(Stream::of).toArray(String[]::new);
+        return idBuilder(allParts);
+    }
+
+    /**
+     * Concatenates all the parts with slashes
+     * @param parts array of strings
+     * @return the concatenated string.
+     */
+    private static String idBuilder(final String... parts) {
+        if (parts != null && parts.length > 0) {
+            final String id = Arrays.stream(parts).map(s -> s.startsWith("/") ? s.substring(1) : s)
```

Comment:
I did think about that, but I was thinking that the use of array of strings would be internal to the codebase (as we normally pass in the full ID from the HTTP level). So it would be our bad code that would include multiple slashes. So not stripping them would (hopefully) catch those errors before they got to the code base. I can strip them, just wanted to outline my thought process.